t/unit-tests/t-reftable-tree.c

```diff
@@ -43,6 +43,7 @@ static void test_tree_search(void)
 		check(nodes[i] == tree_search(values + i, &root, &test_compare, 0));
 	}
 
+	check(!tree_search(values, &root, &test_compare, 0));
```

Comment:
A better title for this commit might be "t-reftable-tree: add test for non-existent key".